base/dcp_client.go

```diff
@@ -460,14 +460,59 @@ func (dc *DCPClient) openStream(vbID uint16, maxRetries uint32) (err error) {
 	return fmt.Errorf("openStream failed to complete after %d attempts, last error: %w", openRetryCount, openStreamErr)
 }
 
-func (dc *DCPClient) rollback(ctx context.Context, vbID uint16) (err error) {
+// rollback changes the metadata of the client for the specified vbucket to mark the new startSeqNo and vBucket UUID.
+func (dc *DCPClient) rollback(ctx context.Context, vbID uint16) error {
```

Comment:
If we hit an error retrieving the failover logs or finding the rollback sequence, should we roll the metadata back to zero prior to returning error and triggering panic?  (so that in the case of persisted metadata, we're not just going to roll back again after restart)